azurerm/resource_arm_virtual_network_gateway.go

```diff
@@ -102,6 +102,19 @@ func resourceArmVirtualNetworkGateway() *schema.Resource {
 				),
 			},
 
+			"generation": {
+				Type:     schema.TypeString,
+				Default:  network.VpnGatewayGenerationGeneration1,
```

Comment:
could we cast this here:

```suggestion
				Default:  string(network.VpnGatewayGenerationGeneration1),
```